base-containers/base/inginious/feedback.py

```diff
@@ -86,6 +86,27 @@ def set_problem_feedback(feedback, problem_id, append=False):
     save_feedback(rdict)
 
 
+def set_tag(tag, value):
+    """ 
+    Set the tag 'tag' to the value True or False. 
+    :param value: should be a boolean 
+    :param tag: should be the id of the tag 
+    """
+    if (value == True):
+        inginious.tests.set_result(tag, True)
+    else:
+        inginious.tests.set_result(tag, False)
+        
+_current_auto_tag = 0
+def tag(value):
```

Comment:
``funcname.variable`` is better for such static variables.
I suggest to prepend ``auto-tag`` with a reserved char to avoid user to set them manually.
Actually, if only a distinction between tags and auto-tags are needed, I would recommend to replace the tag id by a reserved char and a random hash, because static variables will be reset if we call the API using the shell, which is a desired feature for non-Python users.

I'd re-use the definetest code to define a new shell command such as ``tag-result`` or ``tag-set``.